bundles/cif-virtual-catalog/src/main/java/com/adobe/cq/commerce/virtual/catalog/data/impl/CatalogDataResourceProviderManagerImpl.java

```diff
@@ -102,7 +102,7 @@
 
     private EventListener[] observationEventListeners;
 
-    private volatile List<Resource> dataRoots;
+    private final List<Resource> dataRoots = Collections.synchronizedList(new ArrayList<>());
```

Comment:
You could use a CopyOnWriteArrayList instead as I think for this usecase read operations outnumber mutations, right?

Reply:
Indeed, CopyOnWriteArrayList looks like a good fit here. I've changed that.